### PR TITLE
FS Cloud refactor

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Description

Refactor of FS Cloud profile to handle multiple buckets and CBR rules. Added tests for CBRs

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

##### Release notes content
- refactor of fscloud submodule to support multiple buckets
- refactor includes support for all existing bucket configurations
- fix for deploying CBRs with enforcement mode set to enabled, previously this would fail
- added unit tests for CBRs in fscloud submodule to validate they are correctly configured and applied
- added additional outputs including CBR rule IDs which can be useful for testing

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: modules/buckets/outputs.tf [hunk 1]
```diff
@@ -10,3 +10,12 @@ output "bucket_configs" {
   description = "List of bucket config definitions"
   value       = var.bucket_configs
 }
+
+output "bucket_cbr_rules" {
+  description = "List of COS bucket rules"
+  value       = flatten([for _, bucket in module.buckets : bucket.bucket_cbr_rule])
+}
+output "cbr_rule_ids" {
```
Comment (shemau) [inline modules/buckets/outputs.tf:18]: These would read better as `cbr_rules` and `cbs_rule_ids`

**Review by shemau — CHANGES_REQUESTED** [review]
Some changes to variable/output names to make them more consistent.

File: modules/fscloud/version.tf [hunk 2]
```diff
@@ -4,11 +4,4 @@
 terraform {
   required_version = ">= 1.3.0, <1.6.0"
 
-  # Use a flexible range in modules that future proofs the module's usage with upcoming minor and patch versions
-  required_providers {
-    ibm = {
-      source  = "ibm-cloud/ibm"
-      version = ">= 1.56.1, <2.0.0"
-    }
-  }
```
[the remark this answers is not in the document]
Comment (daniel-butler-irl) [reply modules/fscloud/version.tf:13]: I think you are right and I should have added `# tflint-ignore: terraform_unused_required_providers`

File: modules/fscloud/main.tf [hunk 3]
```diff
@@ -35,132 +28,89 @@ module "cos_instance" {
...
-    deletemarker_replication_status = false
-    destination_bucket_crn          = module.buckets.buckets[var.secondary_bucket_name].bucket_crn
-  }
 }
 
-### Configure IAM authorization policy
-
-# Data source to retrieve account ID
-data "ibm_iam_account_settings" "iam_account_settings" {
+module "instance_cbr_rule" {
```
[the remark this answers is not in the document]
Comment (daniel-butler-irl) [reply modules/fscloud/main.tf:77]: I think I need to check this. I thought you could only have one rule per instance. But you can have multiple zones. I could be wrong, checking.....

[the remark this answers is not in the document]
Comment (daniel-butler-irl) [reply modules/fscloud/main.tf:77]: I was wrong, I'll update now

**Review by shemau — APPROVED** [review]
LGTM

**Comment by ocofaigh** [comment]
I'm thinking this is a major release due to the fact the fscloud submodule has been refactored and has no backward compatibility. 
Can you please also include release notes in the PR description.

**Pull request merged**